src/bin/driver.rs

```diff
@@ -549,10 +564,40 @@ mod app {
         c.local.driver_dac[channel as usize].set(current).unwrap()
     }
 
-    #[task(priority = 1, shared=[interlock_handle])]
+    #[task(priority = 1, shared=[interlock_handle, laser_interlock_pin])]
     fn trip_interlock(mut c: trip_interlock::Context) {
         c.shared.interlock_handle.lock(|handle| *handle = None);
-        log::error!("interlock tripped");
+        log::error!("thermostat interlock tripped");
+        c.shared.laser_interlock_pin.lock(|pin| pin.set_low());
+    }
+
+    #[task(priority = 1, shared=[telemetry, settings, laser_interlock_pin], local=[internal_adc])]
+    fn monitor_output(mut c: monitor_output::Context) {
+        let limits = c.shared.settings.lock(|settings| {
+            [
+                settings.low_noise.interlock_current,
+                settings.high_power.interlock_current,
+                settings.low_noise.interlock_voltage,
+                settings.high_power.interlock_voltage,
+            ]
+        });
+        let measurements = c.local.internal_adc.read_all();
+        for (limit, measurement) in limits.iter().zip(measurements.iter()) {
+            if (measurement.0 > *limit)
+                && c.shared
+                    .laser_interlock_pin
+                    .lock(|pin| pin.get_state() == PinState::High)
+            {
+                c.shared.laser_interlock_pin.lock(|pin| pin.set_low());
+                log::info!("Driver output Overvoltage/current condition! {:?} Laser interlock tripped.", measurement.1);
+            }
+        }
+        c.shared.telemetry.lock(|tele| {
```

Comment:
This would also become more readable once the `internal_adc` api matches the `settings.x` api and the `telemetry.monitor` structure.
I'd make `internal_adc` have two functions returning 2-arrays. Then the telemetry becomes easy.
And the iteration as well. Maybe use `flatten()` if it helps.

Also why are the settings called `interlock` and the things here called `limit`?